### ISSUE TITLE
`TypeIs` doesn't properly narrow callable

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#8951]
This code should not produce an error.

```python
from typing import Callable, assert_type
from typing_extensions import TypeIs

def is_callable(obj: object, /) -> TypeIs[Callable[..., object]]: ...

def func1(x: type[int]):
    if is_callable(x):
        assert_type(x, type[int])

def func2[T](x: type[T]):
    if is_callable(x):
        assert_type(x, type[T])

def func3[T](x: T) -> T:
    if not is_callable(x):
        raise ValueError()
    return x
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.380.
